### ISSUE TITLE
pylint does not recognize pytest fixtures

### ISSUE THREAD

**Issue by tslever** [pylint-dev/pylint#9808]
### Bug description

```python
I have written a Python module containing fixtures called `fixtures.py` that are imported by modules with functions that pytest runs. One fixture provides a temporary directory. Another fixture takes the first fixture as a formal parameter, provides the temporary directory, and creates files in the temporary directory. Another fixture provides a logger. Linting `fixtures.py` results in the appropriate warnings below. Linting a module that uses fixture logger results in the appropriate warnings below.
```

### Command used

```shell
find . -path ./env -prune -o -name "*.py" -exec pylint {} +
```

### Pylint output

```python
************* Module fixtures
tests/fixtures.py:59:4: W0621: Redefining name 'temporary_directory' from outer scope (line 31) (redefined-outer-name)
tests/fixtures.py:132:4: W0621: Redefining name 'temporary_directory' from outer scope (line 31) (redefined-outer-name)

tests/test_system.py:14:4: W0613: Unused argument 'logger' (unused-argument)
```

### Expected behavior

I believe linting should not result in such warnings.

### Pylint version

```shell
pylint 3.2.3
astroid 3.2.3
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
```

### OS / Environment

Windows Subsystem for Linux on Windows 10

PRETTY_NAME="Ubuntu 22.04.4 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.4 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy

**Comment by DanielNoord** [comment]
Please have a look at https://github.com/pylint-dev/pylint-pytest.
